File: sample/src/main/java/com/massivedisaster/adal/sample/feature/splash/FragmentSplash.java
/*
 * ADAL - A set of Android libraries to help speed up Android development.
 * Copyright (C) 2017 ADAL.
 *
 * ADAL is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 3 of the License, or any later version.
 *
 * ADAL is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Lesser General Public License along
 * with ADAL. If not, see <http://www.gnu.org/licenses/>.
 */

package com.massivedisaster.adal.sample.feature.splash;

import com.massivedisaster.activitymanager.ActivityFragmentManager;
import com.massivedisaster.adal.fragment.AbstractSplashFragment;
import com.massivedisaster.adal.sample.R;
import com.massivedisaster.adal.sample.feature.home.FragmentHome;

public class FragmentSplash extends AbstractSplashFragment {

    @Override
    protected int layoutToInflate() {
        return R.layout.fragment_splash_screen;
    }

    @Override
    protected void onSplashStarted() {
        onSplashFinish(openHome());
    }

    private OnFinishSplashScreen openHome() {
        return new OnFinishSplashScreen() {
            @Override
            public void onFinish() {
                ActivityFragmentManager.open(getActivity(), ActivitySplashScreen.class, FragmentHome.class);
                getActivity().finish();
            }
        };
    }
}

File: sample-kotlin/src/main/java/com/massivedisaster/adal/samplekotlin/feature/splash/FragmentSplash.kt
/*
 * ADAL - A set of Android libraries to help speed up Android development.
 *
 * Copyright (c) 2018 ADAL
 *
 * Permission is hereby granted, free of charge, to any person obtaining
 * a copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.massivedisaster.adal.samplekotlin.feature.splash

import android.os.Bundle
import com.massivedisaster.adal.fragment.AbstractSplashFragment
import com.massivedisaster.adal.fragment.AbstractSplashFragment.OnFinishSplashScreen
import com.massivedisaster.adal.samplekotlin.R
import com.massivedisaster.adal.samplekotlin.base.activity.ActivityToolbar
import com.massivedisaster.adal.samplekotlin.feature.home.FragmentHome
import com.massivedisaster.afm.ActivityCall

class FragmentSplash : AbstractSplashFragment() {
    override fun getFromBundle(bundle: Bundle) {
        //Intended
    }

    override fun layoutToInflate(): Int = R.layout.fragment_splash_screen

    override fun restoreInstanceState(savedInstanceState: Bundle?) {
        //Intended
    }

    override fun doOnCreated() {
        //Intended
    }

    override fun onSplashStarted() {
        onSplashFinish(openHome())
    }

    private fun openHome(): AbstractSplashFragment.OnFinishSplashScreen = OnFinishSplashScreen {
            ActivityCall.init(activity!!, ActivityToolbar::class, FragmentHome::class)
                    .build()
            activity!!.finish()
        }
}